r/R/dataset.R

```diff
@@ -163,30 +169,59 @@ DatasetFactory$create <- function(sources) {
 #' - `$Finish(schema)`: Returns a `Source`
 #' @rdname Source
 #' @name Source
-#' @seealso [Dataset] for what do do with a `Source`
+#' @seealso [Dataset] for what to do with a `Source`
 #' @export
 Source <- R6Class("Source", inherit = Object,
   active = list(
     #' @description
     #' Return the Source's `Schema`
     schema = function() {
       shared_ptr(Schema, dataset___Source__schema(self))
-    }
+    },
+    #' @description
+    #' Return the Source's type.
+    type = function() dataset___Source__type_name(self)
+  )
+)
+
+#' @name FileSystemSource
```

Comment:
Add a note in the `Source` docs above that FSS is a type of Source. Also should list the methods/active bindings there (the `@description` doclet isn't being used yet).